vshard/storage/init.lua

```diff
@@ -2692,7 +2692,11 @@ local function storage_cfg(cfg, this_replica_uuid, is_reload)
         --
         -- If a master role of the replica is not changed, then
         -- 'read_only' can be set right here.
-        box_cfg.listen = box_cfg.listen or this_replica.uri
+        local this_replicaset_cfg = vshard_cfg.sharding[this_replicaset.uuid]
+        local this_replica_cfg = this_replicaset_cfg.replicas[this_replica.uuid]
+
+        box_cfg.listen = box_cfg.listen or this_replica_cfg.listen or
+                         this_replica_cfg.uri
```

Comment:
Still I afraid box.NULLs in such cases :)

Reply:
Done:
```Diff
@@ -2695,8 +2695,12 @@ local function storage_cfg(cfg, this_replica_uuid, is_reload)
         local this_replicaset_cfg = vshard_cfg.sharding[this_replicaset.uuid]
         local this_replica_cfg = this_replicaset_cfg.replicas[this_replica.uuid]
 
-        box_cfg.listen = box_cfg.listen or this_replica_cfg.listen or
-                         this_replica_cfg.uri
+        if box_cfg.listen == nil then
+            box_cfg.listen = this_replica_cfg.listen
+            if box_cfg.listen == nil then
+                box_cfg.listen = this_replica_cfg.uri
+            end
+        end
         if not box_cfg.replication then
```